fabric-recipe-api-v1/src/testmod/resources/data/minecraft/recipes/iron_ingot_from_smelting_deepslate_iron_ore.json

```diff
@@ -0,0 +1,13 @@
+{
+  "type": "minecraft:smelting",
+  "group": "iron_ingot",
+  "ingredient": {
+    "item": "minecraft:deepslate_iron_ore"
+  },
+  "result": {
+    "item": "minecraft:iron_ingot",
+    "count": 3
```

Comment:
I'm assuming it should be `fabric:count` instead of `count`

Reply:
This is using the Vanilla Codec for processing crafting recipe items with count. There is merit to renaming it in order to differentiate it from Vanilla datapack processing, but personally I prefer the consistency.
I'm open to changing this however, but for now I am partial towards the proposal Shadows-of-Fire mentioned in the NBT thread.

Reply:
Well, I personally agree with your point of view, but fabric adds a suffix to all of its changes to vanilla formats (look at resource conditions and custom ingredients), so consistency should be assessed from fabric's point of view, not vanilla's

Reply:
Thanks for pointing me to these, I will take a look at them. 
I agree that consistency should be prioritised over the mod as a whole, so I will look into building a separate codec to use with `fabric:count` and build and test it over the next few days.

Reply:
How about changing the new result format to `fabric:result`?, Leaving out the normal `result`. 
So something like this:
```
{
  "type": "minecraft:smelting",
  "group": "iron_ingot",
  "ingredient": {
    "item": "minecraft:deepslate_iron_ore"
  },
  "fabric:result": {
    "item": "minecraft:iron_ingot",
    "count": 3
  },
  "experience": 0,
  "cookingtime": 1
}
```

My reasoning is as follows: This change does not simply add to an additional option to an existing format, but instead adds an entirely new way of expressing results. For example, simply naming the (optional) count value `fabric:count` does not imply that a result of the format ```
"result": {
  "item": "minecraft:iron_ingot"
}
```
Would fail to compile in Vanilla (which it would). Renaming to `fabric:result` over `result` would express that this is a fabric-specific datapack file, even in the case where an item count is not provided.

Reply:
How will this work with datagen?

Reply:
Some form of additional functionality would be required for modders to be able to use this functionality with datagen. I do not intend to implement this within this PR.

Reply:
I mean, to what extent is `fabric:result` a realistic option? I would prefer the option that is easier from a datagen point of view, because a large number of recipes are created through datagen. I'm not saying you need to implement datagen support, I'm saying we can't make a choice without datagen